code/__HELPERS/_lists.dm

```diff
@@ -303,9 +303,11 @@
 /proc/typecache_filter_list(list/atoms, list/typecache)
 	RETURN_TYPE(/list)
 	. = list()
-	for(var/atom/atom_checked as anything in atoms)
-		if (typecache[atom_checked.type])
-			. += atom_checked
+
+	if(typecache)
```

Comment:
Have we investigated `/datum/ai_behavior/find_and_set/proc/search_tactic` to determine whether or not a `null` typecache is an error or not? 

At first glance this isn't actually fixing anything, it's just silencing an error, but I don't have enough context. I just have a gut feeling you would want to know if you're accidentally passing a `null` typecache to `typecache_filter_list` because that's indicative of something going wrong

Reply:
I didn't have a good look at why typecache is null, I will do it soon and update with a comment here. I imagined it was out of my depth but can't hurt to have a look.

Reply:
Ok, after having a look at it, it is not the grass patch's fault, the penguin chick mob causes this exception every time it fires the `search_tactic` code. I will if I can cook up a fix for it.

Reply:
Well apparently my branch was out of date, the fix has already been created so I will close the PR